runtime/sema/access.go

```diff
@@ -185,12 +274,102 @@ func (e EntitlementMapAccess) IsLessPermissiveThan(other Access) bool {
 	}
 }
 
+func (e EntitlementMapAccess) Domain() EntitlementSetAccess {
+	var domain map[*EntitlementType]struct{} = make(map[*EntitlementType]struct{})
```

Comment:
Maybe it's worth caching the resulting `EntitlementSetAccess` related to the `domain` and `subdomain` (below) in the `EntitlementMapAccess`